pkg/operator/apiserver/controller/apiservice/apigroup.go

```diff
@@ -101,7 +101,7 @@ func checkDiscoveryForAPIService(restclient rest.Interface, apiService *apiregis
 			defer wg.Done()
 			defer utilruntime.HandleCrash()
 
-			ctx, ctxCancelFn = context.WithTimeout(context.TODO(), 25*time.Second)
+			ctx, ctxCancelFn = context.WithTimeout(context.Background(), 25*time.Second)
```

Comment:
why no ctx from outside? IMO background is wrong for go routines with an wg.Wait just behind in code.

Reply:
fixed, wired context from the outside